File: artem_erp/doc_events/payroll_entry.py
import frappe
from frappe import _
from frappe.utils import get_link_to_form


def check_additional_salary_review_completed(doc):
	if not (doc.company and doc.start_date and doc.end_date):
		return

	draft_reviews = frappe.get_all(
		"Employee Additional Salary Payroll Review",
		filters={
			"company": doc.company,
			"payroll_from_date": ["<=", doc.end_date],
			"payroll_to_date": [">=", doc.start_date],
			"docstatus": 0,
		},
		fields=["name"],
		limit=1,
	)
	if draft_reviews:
		frappe.throw(
			_(
				"Cannot create Salary Slips: Employee Additional Salary Payroll Review {0} is still in Draft or In Review. Please complete and submit the review first."
			).format(get_link_to_form("Employee Additional Salary Payroll Review", draft_reviews[0].name)),
			title=_("Additional Salary Review Required"),
		)


def before_submit(doc, method=None):
	check_additional_salary_review_completed(doc)

File: artem_erp/public/js/payroll_entry.js
// Copyright (c) 2026, Artem Healthtech and contributors
// For license information, please see license.txt

frappe.ui.form.on("Payroll Entry", {
	refresh(frm) {
		toggle_additional_salary_review_button(frm);
	},

	onload_post_render(frm) {
		toggle_additional_salary_review_button(frm);
	},

	company(frm) {
		toggle_additional_salary_review_button(frm);
	},

	start_date(frm) {
		toggle_additional_salary_review_button(frm);
	},

	end_date(frm) {
		toggle_additional_salary_review_button(frm);
	},
});

function toggle_additional_salary_review_button(frm) {
	if (frm.is_new() || frm.doc.docstatus !== 0) {
		frm.remove_custom_button(__("Review Employee Additional Salary"));
		return;
	}

	if (!frm.doc.company || !frm.doc.start_date || !frm.doc.end_date) {
		frm.remove_custom_button(__("Review Employee Additional Salary"));
		return;
	}

	frappe.call({
		method: "artem_erp.artem_erp.doctype.employee_additional_salary_payroll_review.monthly_weekly_payroll_review.check_payroll_review_status",
		args: {
			company: frm.doc.company,
			start_date: frm.doc.start_date,
			end_date: frm.doc.end_date,
		},
		callback: function (r) {
			if (!frm.doc || frm.doc.docstatus !== 0) return;

			frm.remove_custom_button(__("Review Employee Additional Salary"));

			if (r.message && r.message.exists && r.message.docstatus === 0) {
				// Review is in Draft: show custom button without overriding primary action / Save functionality
				frm.add_custom_button(__("Review Employee Additional Salary"), function () {
					if (frm.is_dirty()) {
						frappe.confirm(
							__("You have unsaved changes. Do you want to save before reviewing Additional Salaries?"),
							() => {
								frm.save().then(() => {
									frappe.set_route("Form", "Employee Additional Salary Payroll Review", r.message.name);
								});
							},
							() => {
								frappe.set_route("Form", "Employee Additional Salary Payroll Review", r.message.name);
							}
						);
					} else {
						frappe.set_route("Form", "Employee Additional Salary Payroll Review", r.message.name);
					}
				});
			}
		},
	});
}
